File: M3Tools/Dialogs/Customer/CustomerBasedDialogs.vb
Imports System.Windows.Forms

Public Class CustomerBasedDialogs
	Public Function AddCustomer() As DialogResult
		Return (New Dialogs.AddCustomerDialog()).ShowDialog
	End Function

	Public Function EditCustomer(customer As Types.Customer) As DialogResult
		Return (New Dialogs.EditCustomerDialog() With {.Customer = customer}).ShowDialog
	End Function
End Class

File: M3Tools/Dialogs/Customer/CustomerBasedDialogs.cs
using System.Windows.Forms;

namespace SPPBC.M3Tools
{
	/// <summary>
	/// A component that houses all customer based dialogs forms
	/// </summary>
    public partial class CustomerBasedDialogs
    {
		/// <summary>
		/// Open a dialog box for adding a customer
		/// </summary>
		/// <returns></returns>
        public DialogResult AddCustomer()
        {
            return new Dialogs.AddCustomerDialog().ShowDialog();
        }

		/// <summary>
		/// Open a dialog box for editing a customer
		/// </summary>
		/// <param name="customer">The customer to edit</param>
		/// <returns></returns>
        public DialogResult EditCustomer(Types.Customer customer)
        {
            return new Dialogs.EditCustomerDialog(customer).ShowDialog();
        }
    }
}
